MiFoDB
=======================================

Data supporting publication: MiFoDB, a workflow for microbial food metagenomic characterization, enables high-resolution analysis of fermented food microbial dynamics
+++++++++++++++

This upload contains several datasets to support the publication "MiFoDB, a workflow for microbial food metagenomic characterization, enables high-resolution analysis of fermented food microbial dynamics", including mapping databases and other relevant data. In addition, raw reads for each sample and MAGs are available on SRA SubmissionID SUB14310273 and BioProjectID PRJNA1092721.

Zenodo Content Directory
+++++++++++++++
Link to Zenodo: https://zenodo.org/records/13766023

**MiFoDB_gamma_v1.zip**: Includes updated the fasta, stb, and gene file and mapping file for a prokaryote database with 1294 representative genomes from this data set plus data from Carlino et al. 2024
 * MiFoDB_gamma_v1_prok.fasta: fasta file of all 1294 concatenated representative prokaryote genomes in database. 
 * MiFoDB_gamma_v1_prok.stb: scaffold-to-bin file for the 1294 prokaryote genomes in database.
 * MiFoDB_gamma_v1_prok.genes.fna: prodigal output for 1294 prokaryote genomes reference in database. 
 * MiFoDB_beta_v3_euk.fasta: fasta file of all 99 concatenated representative eukaryote microorganism genomes in database. 
 * MiFoDB_beta_v3_euk.stb: scaffold-to-bin file 99 for the eukaryote microorganism genomes in database.
 * MiFoDB_beta_v1_sub.fasta: fasta file of all 7 concatenated representative substrate genomes in database. 
 * MiFoDB_beta_v1_sub.stb: scaffold-to-bin file for the 7 substrate genomes in database. 

**MiFoDB_gamma_v1_Reference.csv**: .csv of all genomes included in MiFoDB_gamma_v1.

**MiFoDB_beta_v3.zip**: Includes the following files updated with new prokaryotic (939 representative) and eukaryotic (99 representative) genomes:
 * MiFoDB_beta_v3_prok.fasta: fasta file of all 939 concatenated representative prokaryote genomes in database. 
 * MiFoDB_beta_v3_prok.stb: scaffold-to-bin file for the 939 prokaryote genomes in database. 
 * MiFoDB_beta_v3_prok.genes.fna: prodigal output for 939 prokaryote genomes reference in database. 
 * MiFoDB_beta_v3_euk.fasta: fasta file of all 99 concatenated representative eukaryote microorganism genomes in database. 
 * MiFoDB_beta_v3_euk.stb: scaffold-to-bin file for the 99 eukaryote microorganism genomes in database.
 * MiFoDB_beta_v1_sub.fast: fasta file of all 7 concatenated representative substrate genomes in database. 
 * MiFoDB_beta_v1_sub.stb: scaffold-to-bin file for the 7 substrate genomes in database. 

**MiFoDB_beta_v3_Reference.csv**: .csv of all genomes included in MiFoDB_beta_v3.

**MiFoDB_beta_v2.zip**: Includes the following files used in profiling of results published in the paper:
 * MiFoDB_beta_v2_prok.fasta: fasta file of all concatenated representative prokaryote genomes in database. 
 * MiFoDB_beta_v2_prok.stb: scaffold-to-bin file for the prokaryote genomes in database.
 * MiFoDB_beta_v2_prok.genes.fna: prodigal output for prokaryote genomes reference in database.
 * MiFoDB_beta_v2_euk.fasta: fasta file of all concatenated representative eukaryote microorganism genomes in database. 
 * MiFoDB_beta_v2_euk.stb: scaffold-to-bin file for the eukaryote microorganism genomes in database. 
 * MiFoDB_beta_v1_sub.fasta: fasta file of all concatenated representative substrate genomes in database. 
 * MiFoDB_beta_v1_sub.stb: scaffold-to-bin file for the substrate genomes in database. 

**MiFoDB_beta_v2_Reference.csv**: .csv of all genomes included in MiFoDB_beta_v2.

**MiFoDB_beta_v2_allprokgenomes.zip**: directory of all prokaryote genomes as .fa files used in the final version of MiFoDB_beta_v2.

**MiFoDB_beta_v2_alleukgenomes.zip**: directory of all eukaryote genomes as .fa files used in the final version of MiFoDB_beta_v2.

**Caffreyetal2024_MiFoDB_MAGs.zip**: all MAGs generated in this study.

**EBC_087.zip**: Fastq files for EBC_087_S160_L003_R1 and EBC_087_S160_L003_R2 used in https://mifodb.readthedocs.io/ example.

For a full list of genomes included in each database, see the `MiFoDB reference <https://docs.google.com/spreadsheets/d/1PHRlb9YwKiwpVk8ChozBZbFYCA-VL3EXJTIPI-TI04A/edit?usp=sharing>`_ and check the database column. 

